test/lib/check_rack_attack_test.rb

```diff
@@ -0,0 +1,31 @@
+require 'test_helper'
+
+class ThrottlingTest < ActionDispatch::IntegrationTest
+  setup do
+    Rails.cache.clear
+  end
+
+  test "should throttle excessive requests to /api/graphql" do
+    limit = 100
+
+    limit.times do
+      post api_graphql_path
+      assert_response :unauthorized
+    end
+
+    get api_graphql_path
+    assert_response :too_many_requests
+  end
```

Comment:
@jayjay-w , I also suggest that you use `stub_config` here (there are examples in code) so that you don't need to test with 100 requests (you can test with 3, for example)... I think this is impacting how long the tests take.

Reply:
Thank you, this is fixed as well!